x-pack/legacy/plugins/apm/public/components/app/Settings/AgentConfigurations/AgentConfigurationCreateEdit/SettingsPage/SettingFormRow.tsx

```diff
@@ -0,0 +1,178 @@
+import { SelectWithPlaceholder } from '../../../../../shared/SelectWithPlaceholder';
+
+function FormRow({
+  setting,
+  value,
+  onChange
+}: {
+  setting: ConfigSettingDefinition;
+  value?: string;
+  onChange: (key: string, value: string) => void;
+}) {
+  const defaultInputPlaceholder = i18n.translate(
+    'xpack.apm.agentConfig.defaultInputPlaceholder',
+    {
+      defaultMessage: 'Set {settingsLabel}',
+      values: { settingsLabel: setting.label }
+    }
+  );
+
+  switch (setting.type) {
+    case 'text': {
+      return (
+        <EuiFieldText
+          placeholder={setting.placeholder || defaultInputPlaceholder}
+          value={value || ''}
+          onChange={e => onChange(setting.key, e.target.value)}
+        />
+      );
+    }
+
+    case 'integer': {
+      return (
+        <EuiFieldNumber
+          placeholder={setting.placeholder || defaultInputPlaceholder}
+          value={(value as any) || ''}
+          min={setting.min}
+          max={setting.max}
+          onChange={e => onChange(setting.key, e.target.value)}
+        />
+      );
+    }
+
+    case 'select': {
+      return (
+        <SelectWithPlaceholder
+          placeholder={setting.placeholder}
+          options={setting.options}
+          value={value}
+          onChange={e => onChange(setting.key, e.target.value)}
+        />
+      );
+    }
+
+    case 'boolean': {
+      return (
+        <SelectWithPlaceholder
+          placeholder={setting.placeholder}
+          options={[{ text: 'true' }, { text: 'false' }]}
+          value={value}
+          onChange={e => onChange(setting.key, e.target.value)}
+        />
+      );
+    }
+
+    case 'amountAndUnit': {
+      const { amount, unit } = amountAndUnitToObject(value ?? '');
+
+      return (
+        <EuiFlexGroup gutterSize="none">
+          <EuiFlexItem grow={false}>
+            <EuiFieldNumber
+              placeholder={setting.placeholder || defaultInputPlaceholder}
+              value={amount as any}
+              onChange={e =>
+                onChange(
+                  setting.key,
+                  amountAndUnitToString({ amount: e.target.value, unit })
+                )
+              }
+            />
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <SelectWithPlaceholder
+              placeholder={i18n.translate('xpack.apm.unitLabel', {
+                defaultMessage: 'Select unit'
+              })}
+              value={unit}
+              options={setting.units.map(text => ({ text }))}
+              onChange={e =>
+                onChange(
+                  setting.key,
+                  amountAndUnitToString({ amount, unit: e.target.value })
+                )
+              }
+            />
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      );
+    }
+
+    default:
+      // @ts-ignore
```

Comment:
Since all the cases currently have been handled it complains about the default case (because it's not needed)
![image](https://user-images.githubusercontent.com/209966/76969186-53946b00-692a-11ea-8673-dd07d3a591b7.png)

I'll change it to:
```ts
      throw new Error(
        `Unknown type "${(setting as ConfigSettingDefinition).type}"`
      );
```